torch/amp/grad_scaler.py

```diff
@@ -334,6 +343,9 @@ def unscale_(self, optimizer: torch.optim.Optimizer) -> None:
         assert self._scale is not None
         inv_scale = self._scale.double().reciprocal().float()
         found_inf = torch.full((), 0.0, dtype=torch.float32, device=self._scale.device)
+        if optimizer.param_groups[0]["params"] is not None:
+            # if optimizer has params as DTensor, we initialize found_inf as DTensor by DTensor.new_full()
+            found_inf = optimizer.param_groups[0]["params"][0].new_full((), 0.0)
```

Comment:
this changed the code path for plain torch.Tensor. maybe consider gating with if `"DTensor" in grads[0].__class__.__name__`

`grads[0].new_full()` also does the job since `found_inf` is computed from `grads`